### PR TITLE
feat: project num 4

### PR DESCRIPTION
**Pull request by ShaunSHamilton**
Adds _Learn Solana's Token Program by Minting a Fungible Token_ practice project - the 4th project

TODO:

- [x] Add `package.json` to project root
- [x] Test tests

### CODE REVIEW

File: config/projects.json [hunk 1]
```diff
@@ -31,6 +31,17 @@
...
+    "dashedName": "learn-solanas-token-program-by-minting-a-fungible-token",
+    "isIntegrated": false,
+    "description": "In this course, you will learn how to use Solana's token program by minting a fungible token.",
+    "isPublic": true,
+    "currentLesson": 1,
+    "runTestsOnWatch": true,
+    "numberOfLessons": 65
+  },
   {
     "id": 3,
```
Comment (moT01) [inline config/projects.json:46]:
```suggestion
    "id": 4,
```

**Review by moT01 — COMMENTED** [review]
Alright @ShaunSHamilton, I made it through this - it's pretty awesome 🎉 everything was easy to understand and it was a lot of fun 👍 I will list everything I wrote down, it's mostly pretty minor - make changes at your discretion.

- Remove the `_answer` folder?

step 6:
- I entered `solana-keygen new`, then created the `wallet.json` file (empty) and the tests pass. I had to go back to see what I was supposed to do. Maybe add a test to make sure the `wallet.json` file has something in it

step 7:
- pretty minor - I added `const mintAuthority` and it auto imported `mintAuthority` from `utils.js`. Test didn't pass until I deleted it. Could maybe add a test that says "You should not have that imported" or something - not sure. "You should only have one `mintAuthority` variable" ?

step 9:
- I did this wrong and a problem came up later (see step 13). I had `const mint = await new createMint()` - note the `new`, it is not a constructor and the test passed. 

step 11:
- Had to run the tests manually, could maybe add a note about it. Although, they should have that figured out at this point.

step 13:
- I like the expandable note, I may have to utilize that. The text for that area ran way off to the right creating a horizontal scroll bar. Maybe an issue for fccOS
- I made the mistake on step 9 and the output here gave me a syntax error and did not include the base58 log, but the test passed. Could maybe add a test to make sure the correct thing is logged.

step 20:
- it worked for me, but could maybe add a test like I suggested on 13 to check that the output includes that log

step 24:
- Third time I'm creating a connection. Could maybe be a little more vague with the instructions. Really nit-picking, probably just leave it.

step 29:
- Just an observation - it says `solana address -k` - one thing I tried to start doing (perhaps not all that consistently), about midway through the RDMS, is use the full flag - e.g. `solana address --keypair`. I think it will help users better understand what the command is doing and can associate `-k` with a word, `keypair`, when they figure out the short flag.

- Could maybe use a test to check output again here in case something isn't correct

step 30:
- a `amount` -> an `amount` ?

step 31:
- fourth time creating a connection - could be more vague with instructions. 

step 32:
- instructions don't mention to await - campers should know though. It is mentioned in the tests.

step 33, 34:
- could have a test to check the output

step 39:
- test to check output? 
- Also, I had an issue here - I put a space at the end of my command and the test failed. I believe when I'm looking at the lastCommand, I trim the spaces off the ends and replace instances of one or more space with a single space.  
Screen Shot 2023-02-08 at 11 32 42 AM

step 40:
- the third test ("The total supply...") hung for a while, eventually it passed - not sure, it went quick on another attempt.
- could possibly change that test to check that they have >= 3_000... in case the tests don't pass right away and they have to mint another.

step 51:
- seems like it needs a slight rewording in the instructions? public key is in there twice, just a little confused
- add a test to check the output? I entered wrong command and it passed.

- I was having issues... I think I figured out what to do. The instructions say to run `get-token-account <wallet.json_pubKey>`. So I did, and got some info, but none of it included the public key I needed for the transfer (I don't think). I added `console.log(tokenAddress.toBase58());` to the `get-token-address.js` file and got a key I could use for the transfer and I think it worked as intended. Unless I messed up along the way, you may want to add that log in that file at some point or something so it gets logged and users can copy it for the transfer.

**Comment by ShaunSHamilton** [comment]
@moT01 Thank you for taking the time to review this.

- Step 6: Nice catch. Campers should remember this from previous projects, but I can see it easily overlooked.
- > I like the expandable note, I may have to utilize that. The text for that area ran way off to the right creating a horizontal scroll bar. Maybe an issue for fccOS
  - Already fixed on fccOS. Just need to update it here.
- Step 29: > I think it will help users better understand what the command is doing and can associate -k with a word, keypair, when they figure out the short flag.
  - I will do this here (and probably in general). The main reason I do not consider it a problem is the first project has Campers go through all the motions of `solana --help`, `solana program --help`, `solana airdrop --help`, etc. So, I a) expect them to remember, b) expect them to know how to see the flags
- Step 30: 🤫 
- > I believe when I'm looking at the lastCommand, I trim the spaces off the ends and replace instances of one or more space with a single space.  
  - I have every other test like this use `assert.include` for this reason, but did an `assert.equal` here for some reason 🤷‍♂️ 
- > could possibly change that test to check that they have >= 3_000... in case the tests don't pass right away and they have to mint another.
  - Good idea. Thanks
- > add a test to check the output? I entered wrong command and it passed.
  - I do not know why it would do that. I should probably make testing the output more of a priority.
- Step 51: ?? The output should include:
```typescript
Account {
  address: PublicKey
}
```

**Comment by ShaunSHamilton** [comment]
Merging into `next`. Leaving `_answer` folders until `next` is merged into `main`.

**Pull request merged**